src/DesaiHardeningStressUpdate.C

```diff
@@ -1314,37 +1232,23 @@ DesaiHardeningStressUpdate::setGamaValue(const std::vector<Real> & stress_params
   std::vector<Real> lamda(3);
   RankTwoTensor eigv = RankTwoTensor();
 
-  const Real dd = 3.14159 * _dip / 180.0;
-  const Real ss = 3.14159 * _strike / 180.0;
-
-  const Real c1 = std::cos(ss);
-  const Real c2 = std::cos(dd);
-
-  const Real s1 = std::sin(ss);
-  const Real s2 = std::sin(dd);
-
-  rotation(0, 0) = c1;  // R11
-  rotation(1, 0) = s1;  // R21
-  rotation(2, 0) = 0.0; // R31
+  // const auto e1 = _localCoordinateSystem.e1();
+  // const auto e2 = _localCoordinateSystem.e2();
+  const auto e3 = _localCoordinateSystem.e3();
 
-  rotation(0, 1) = -c2 * s1; // R12
-  rotation(1, 1) = c1 * c2;  // R22
-  rotation(2, 1) = s2;       // R32
+  (stress_now).symmetricEigenvaluesEigenvectors(lamda, eigv);
 
-  rotation(0, 2) = s2 * s1;  // R13
-  rotation(1, 2) = -c1 * s2; // R23
-  rotation(2, 2) = c2;       // R33
+  const Real cosb = eigv(0, 0) * e3(2) + eigv(1, 0) * e3(1) + eigv(2, 0) * e3(0);
```

Comment:
here we need the cosine of the angle between first principal stress (eigv(x,0)) and the normal vector to the beddeing (e3), so i thinkc it should be: 
const Real cosb = eigv(0, 0) * e3(0) + eigv(1, 0) * e3(1) + eigv(2, 0) * e3(2);